main.cpp

```diff
@@ -376,6 +330,38 @@ DoubleType::~DoubleType()
     delete value;
 }
 
+DoubleType& DoubleType::pow(double num)
+{
+    powInternal(num);
+    return *this;
+}
+
+DoubleType& DoubleType::pow(const DoubleType& dtype)
+{
+    powInternal(static_cast<double>(dtype));
+    return *this;
+}
+
+DoubleType& DoubleType::pow(const FloatType& ftype)
+{
+    double arg = static_cast<double>(ftype);
+    powInternal(arg);
+    return *this;
+}
+
+DoubleType& DoubleType::pow(const IntType& itype)
+{
+    double arg = static_cast<double>(itype);
+    powInternal(arg);
+    return *this;
+}
```

Comment:
What does `powInternal` return?
Why aren't you returning what that function returns?
You should, doing so will eliminate a few lines of repetitive code.
- [ ] revise